snapshots/src/lp-snapshot.ts

```diff
@@ -79,57 +108,95 @@ export async function generateLpSnapshot(
+            eventStartBlock,
+            endBlockBeingUsed: endBlock,
+            diffBetweenNowAndEndBlock: currentBlock - endBlock,
         });
 
-        //Ensure that the market occured within the blocks being checked
-        if (startBlock !== null && endBlock > startBlock) {
-            const samples: number[] = [];
-            for (
-                let block = startBlock;
-                block <= endBlock;
-                block += blocksPerSample
-            ) {
-                samples.push(block);
-            }
-
-            console.log(`Using ${market.howToCalculate} calculation`);
-
-            const tokensPerSample = calculateTokensPerSample(
-                market,
-                samples.length,
-                blocksPerSample,
+        // roughly, if our systems can handle ~150 samples per epoch
+        // then we should take reward_market_start & reward_market_end diff and divide by 150
+        if (rewardMarketStartBlock && rewardMarketEndBlock) {
+            blocksPerSample = calculateSamplesPerEvent(
+                rewardMarketStartBlock,
+                rewardMarketEndBlock,
+                NUMBER_OF_SAMPLES_PER_MARKET,
             );
-            console.log(`Using ${tokensPerSample} tokens per sample`);
             console.log(
-                `Using ${tokensPerSample / blocksPerSample} tokens per block`,
+                `Reward market start and end block exist. Custom sample size: ${blocksPerSample}`,
             );
 
-            console.log(
-                `Diff between now and endBlock is ${
-                    currentBlock - endBlock
-                } blocks. (43,200 = 1 day; 1,800 = 1 hour; 30 = 1 minute)`,
-            );
+            if (eventStartBlock) {
+                validateEventStartBlock(
+                    rewardMarketStartBlock,
+                    eventStartBlock,
+                    rewardMarketEndBlock,
+                );
+                console.log("market maker of market with event", marketMaker);
+            }
+        }
+
+        const arrayOfSamples = createArrayOfSamples(
+            startBlock,
+            endBlock,
+            eventStartBlock,
+            blocksPerSample,
+        );
 
-            // get liquidity state across many blocks for a market
+        // console.log(`arrayOfSamples length: ${arrayOfSamples.length}`);
+
+        arrayOfSamples.forEach(async (samples, idx, arr) => {
             const liquidityAcrossBlocks =
                 await calculateValOfLpPositionsAcrossBlocks(
                     marketMaker,
                     samples,
                 );
 
+            console.log(`number of samples: ${samples.length}`);
+            const marketHasAnEvent = arr.length === 2;
+            // if there are two arrays of blocks, the [1] blocks must be during the event
+            let weight = 1;
+            if (
+                marketHasAnEvent &&
+                typeof market.preEventPercent === "number"
```

Comment:
Should we add error handling here? What would happen if it wasn't a number or was null, should we throw an error and exit?

Reply:
@dylangolow  It will not be a number or null if there is no event. When that happens, we calculate tokens per sample using 100% of the market's amount. However, we do need to add a check for the case when someone adds an event, but does not specify a `preEventPercent`. Currently, it would just allocate 100% of the market amount for the preEvent blocks and another 100% for the event blocks. Added validation and pushed